clients/src/test/java/org/apache/kafka/clients/ClusterConnectionStatesTest.java

```diff
@@ -310,9 +310,6 @@ public void testMultipleIPsWithUseAll() throws UnknownHostException {
         connectionStates.connecting(nodeId1, time.milliseconds(), hostTwoIps, ClientDnsLookup.USE_ALL_DNS_IPS);
         InetAddress addr2 = connectionStates.currentAddress(nodeId1);
```

Comment:
Relying on "specify" number of IPs is still not stable for this tests (for example, the number of resolved IPs becomes 1). It seems to me this test should check whether the number used addresses (after building many connections) is equal to number of resolved addresses.